WooCommerce/Classes/ViewRelated/Upgrades/UpgradesViewModel.swift

```diff
@@ -0,0 +1,187 @@
+        return sanitizedName
+    }
+
+    /// Returns a plan specific details information.
+    ///
+    static func getPlanInfo(from plan: WPComSitePlan) -> String {
+        let daysLeft = daysLeft(for: plan)
+        let planDuration = planDurationInDays(for: plan)
+
+        if plan.isFreeTrial {
+            if daysLeft > 0 {
+                return Localization.freeTrialPlanInfo(planDuration: planDuration, daysLeft: daysLeft)
+            } else {
+                return Localization.trialEndedInfo
+            }
+        }
+
+        let planName = getPlanName(from: plan)
+        guard let expireDate = plan.expiryDate else {
+            return ""
+        }
+
+        let expireText = DateFormatter.mediumLengthLocalizedDateFormatter.string(from: expireDate)
+        return Localization.planInfo(planName: planName, expirationDate: expireText)
+    }
+
+    /// Only allow to upgrade the plan if we are on a free trial.
+    ///
+    static func getUpgradeNowButtonVisibility(from plan: WPComSitePlan) -> Bool {
+        plan.isFreeTrial
+    }
+
+    /// Returns a site plan duration in days.
+    ///
+    static func planDurationInDays(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        guard let subscribedDate = plan.subscribedDate?.startOfDay(timezone: .current),
+              let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let duration = Calendar.current.dateComponents([.day], from: subscribedDate, to: expiryDate).day ?? 0
+        return duration
+    }
+
+    /// Returns how many days site  plan has left.
+    ///
+    static func daysLeft(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        let today = Date().startOfDay(timezone: .current)
+        guard let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let daysLeft = Calendar.current.dateComponents([.day], from: today, to: expiryDate).day ?? 0
+        return daysLeft
+    }
+}
+
+// MARK: Definitions
+private extension UpgradesViewModel {
+    enum Localization {
+        static let freeTrial = NSLocalizedString("Free Trial", comment: "Plan name for an active free trial")
+        static let trialEnded = NSLocalizedString("Trial ended", comment: "Plan name for an expired free trial")
+        static let trialEndedInfo = NSLocalizedString("Your free trial has ended and have limited access to all the features. Subscribe to eCommerce now.",
+                                                      comment: "Info details for an expired free trial")
+        static func freeTrialPlanInfo(planDuration: Int, daysLeft: Int) -> String {
+            let format = NSLocalizedString("You are in the %1d-day free trial. The free trial will end in %2d days. " +
```

Comment:
This format adds an extra space to the number of days until the trial ends. (Example: `The free trial will end in  9 days.` with two spaces before the number `9`.) Using `n$` for the position should fix the extra space, like this:

```suggestion
            let format = NSLocalizedString("You are in the %1$d-day free trial. The free trial will end in %2$d days. " +
```